### PR TITLE
Multiagent simplerl

### PR DESCRIPTION
**Pull request by andrewcoh**
### Proposed change(s)

Describe the changes made in this PR.

### Types of change(s)

- [ ] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Breaking change
- [ ] Documentation update
- [ ] Other (please describe)

### Checklist
- [ ] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/main/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [ ] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/main/docs) (if applicable)
- [ ] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/main/docs/Migrating.md) (if applicable)

### CODE REVIEW

File: ml-agents/mlagents/trainers/tests/torch/test_simple_rl.py [hunk 1]
```diff
@@ -37,11 +42,19 @@
 
 PPO_TORCH_CONFIG = ppo_dummy_config()
 SAC_TORCH_CONFIG = sac_dummy_config()
+COMA_TORCH_CONFIG = coma_dummy_config()
 
 # tests in this file won't be tested on GPU machine
 pytestmark = pytest.mark.check_environment_trains
 
 
+@pytest.mark.parametrize("action_sizes", [(0, 1), (1, 0)])
```
Comment (vincentpierre) [inline ml-agents/mlagents/trainers/tests/torch/test_simple_rl.py:51]: Can this be tested for combinations of rank 1, 2 and 3 observations and with and LSTM config? (To make sure it does not crash at least)

File: ml-agents/mlagents/trainers/tests/simple_test_envs.py [hunk 2]
```diff
@@ -314,6 +314,185 @@ def _make_batched_step(
         return (decision_step, terminal_step)
 
 
+class MultiAgentEnvironment(BaseEnv):
```
[the remark this answers is not in the document]
Comment (andrewcoh) [reply ml-agents/mlagents/trainers/tests/simple_test_envs.py:317]: Yeah, it is a pretty horrible thing

File: ml-agents/mlagents/trainers/tests/torch/test_simple_rl.py (hunk 1 above)
Comment (ervteng) [reply ml-agents/mlagents/trainers/tests/torch/test_simple_rl.py:51]: Added some tests for LSTM, variable length obs, and visual

File: ml-agents/mlagents/trainers/tests/simple_test_envs.py [hunk 3]
```diff
@@ -314,6 +314,212 @@ def _make_batched_step(
...
+                    num_var_len,
+                    vis_obs_size,
+                    vec_obs_size,
+                    var_len_obs_size,
+                    action_sizes,
+                )
+                self.dones[name_and_num] = False
+                self.envs[name_and_num].reset()
+                # HACK
+                self.behavior_spec = self.envs[name_and_num].behavior_spec
```
Comment (vincentpierre) [inline ml-agents/mlagents/trainers/tests/simple_test_envs.py:362]: This dos not need to be in the loop

**Review by vincentpierre — APPROVED** [review]

File: ml-agents/mlagents/trainers/tests/simple_test_envs.py (hunk 3 above)
Comment (ervteng) [reply ml-agents/mlagents/trainers/tests/simple_test_envs.py:362]: Moved

**Pull request merged**